client/concordclient/src/concord_client.cpp

```diff
@@ -232,4 +232,10 @@ void ConcordClient::getSnapshot(const StateSnapshotRequest& request, std::shared
   rss_->readSnapshotStream(rss_request, remote_queue);
 }
 
+bool ConcordClient::isClientPoolHealthy() {
+  if (client_pool_ and client_pool_->HealthStatus() == concord::concord_client_pool::PoolStatus::NotServing)
```

Comment:
With this condition and return value placement, it looks like `isClientPoolHealthy` will return `true` if `client_pool_ is null. Is this intentional? If so, can you comment on the reasoning for this behavior?

(Sorry if I am missing or overlooking any relevant context for this behavior).

Reply:
Thank you for this revision.